### ISSUE TITLE
Fan Issues:  Raw_Key cannot be nulled

### ISSUE THREAD

**Issue by squareflips** [prasad-edlabadka/homebridge-tuya-ir#81]
Hi, 

This plugin is awesome and have been using it for a few months now.  However I noticed that a few days ago, it stopped working (aircon, tv, and fans).  Updating the plugin a few days back seemed to get almost everything back except Fan devices.  I can no longer control my electric fans from Homekit.  I get these errors:

[9/17/2022, 5:58:18 PM] [TuyaIR] Failed to change Fan speed due to error raw_key can not be null
[9/17/2022, 5:58:18 PM] [TuyaIR] Failed to change Fan status due to error raw_key can not be null

Any advice?

**Comment by prasad-edlabadka** [comment]
Please try the latest version. it should fix the problem.

**Comment by squareflips** [comment]
I am currently using v1.0.8

**Comment by prasad-edlabadka** [comment]
I am not able to reproduce this error. I get proper response
```
[18/09/2022, 10:55:37] [TuyaIR] Setting DIY Commands for Fan as {"power":1,"speed":9367,"swing":9362}
[18/09/2022, 10:55:59] [TuyaIR] Calling endpoint https://openapi.tuyaeu.com/v1.0/infrareds/xxx/remotes/xxx/raw/command with payload {"raw_key":1}
[18/09/2022, 10:56:00] [TuyaIR] API call successful. {"result":true,"success":true,"t":1663478760572,"tid":"xxx"}
[18/09/2022, 10:56:00] [TuyaIR] Fan is now On
```

**Comment by prasad-edlabadka** [comment]
Can you please put home bridge in DEBUG mode and capture logs?

**Comment by iwebworx** [comment]
hi @prasad-edlabadka

[Playpen Area B340] The image snapshot handler for the given accessory is slow to respond! See https://homebridge.io/w/JtMGR for more info.
**[10/5/2022, 4:39:00 AM] [TuyaIR] Calling endpoint https://openapi.tuyaus.com/v1.0/infrareds/ebf59a7721ddda73f5hcbh/remotes/eb140787d112052bd6vyb2/raw/command with payload {"raw_key":null}
[10/5/2022, 4:39:01 AM] [TuyaIR] API call successful. {"code":1102,"msg":"raw_key can not be null","success":false,"t":1664915947282,"tid":"97ac931d442411eda5021297425bcc23"}
[10/5/2022, 4:39:01 AM] [TuyaIR] Failed to change Fan status due to error raw_key can not be null**
[10/5/2022, 4:39:14 AM] [Camera FFmpeg] [Playpen Area] Fetching snapshot took 19.732 seconds.

**Comment by prasad-edlabadka** [comment]
Looks like Fan commands are not retrieved correctly. Can you please send complete logs from home bridge startup? There is a debug log statement that logs retrieved IR commands.

**Comment by iwebworx** [comment]
[homebridge-tuya-ir.log.txt](https://github.com/prasad-edlabadka/homebridge-tuya-ir/files/9746864/homebridge-tuya-ir.log.txt)

@prasad-edlabadka thanks in advance

**Comment by prasad-edlabadka** [comment]
Fixed in v1.0.10. Please test.

**Comment by squareflips** [comment]
Fan now works! Thanks a bunch!
